core/src/banking_stage/worker.rs

```diff
@@ -0,0 +1,626 @@
+        let committer = Committer::new(
+            None,
+            replay_vote_sender,
+            Arc::new(PrioritizationFeeCache::new(0u64)),
+        );
+        let consumer = Consumer::new(committer, recorder, QosService::new(1), None);
+
+        let (_local_node, cluster_info) = new_test_cluster_info(None);
+        let cluster_info = Arc::new(cluster_info);
+        let forwarder = Forwarder::new(
+            poh_recorder.clone(),
+            bank_forks,
+            cluster_info,
+            Arc::default(),
+            Arc::default(),
+        );
+
+        let (consume_sender, consume_receiver) = unbounded();
+        let (forward_sender, forward_receiver) = unbounded();
+        let (consumed_sender, consumed_receiver) = unbounded();
+        let (forwarded_sender, forwarded_receiver) = unbounded();
+        let worker = Worker::new(
+            consume_receiver,
+            consumer,
+            consumed_sender,
+            forward_receiver,
+            ForwardOption::ForwardTransaction,
+            forwarder,
+            forwarded_sender,
+            poh_recorder.read().unwrap().new_leader_bank_notifier(),
+        );
+
+        (
+            TestFrame {
+                mint_keypair,
+                genesis_config,
+                bank,
+                ledger_path,
+                entry_receiver,
+                poh_recorder,
+                poh_simulator,
+                replay_vote_receiver,
+                consume_sender,
+                consumed_receiver,
+                forward_sender,
+                forwarded_receiver,
+            },
+            worker,
+        )
+    }
+
+    #[test]
+    fn test_worker_consume_no_bank() {
+        let (test_frame, worker) = setup_test_frame();
+        let TestFrame {
+            mint_keypair,
+            genesis_config,
+            bank,
+            consume_sender,
+            consumed_receiver,
+            ..
+        } = &test_frame;
+        let worker_thread = std::thread::spawn(move || worker.run());
+
+        let pubkey1 = Pubkey::new_unique();
+
+        let transactions = sanitize_transactions(vec![system_transaction::transfer(
+            mint_keypair,
+            &pubkey1,
+            1,
+            genesis_config.hash(),
+        )]);
+        let bid = TransactionBatchId::new(0);
+        let id = TransactionId::new(0);
+        let work = ConsumeWork {
+            batch_id: bid,
+            ids: vec![id],
+            transactions,
+            max_age_slots: vec![bank.slot()],
+        };
+        consume_sender.send(work).unwrap();
+        let consumed = consumed_receiver.recv().unwrap();
+        assert_eq!(consumed.work.batch_id, bid);
+        assert_eq!(consumed.work.ids, vec![id]);
+        assert_eq!(consumed.work.max_age_slots, vec![bank.slot()]);
+        assert_eq!(consumed.retryable_indexes, vec![0]);
+
+        drop(test_frame);
+        let _ = worker_thread.join();
```

Comment:
nit: better off calling `.join().unwrap()`, so that unexpected worker_thread panic should cause these tests fail.

Reply:
Yeah - nice catch! Was forgetting the join-handle returns a result on the thread which wraps the result. Stupidly was thinking unwrapping here would panic because the worker result is an error on a disconnected channel (expected).

Reply:
b3826921217ad427ff0e8074d40afb45af4213ef